src/tools/namespace_tool_core.cpp

```diff
@@ -125,30 +125,52 @@ int NameSpaceToolCore::GetFileSegments(const std::string& fileName,
     return 0;
 }
 
-int NameSpaceToolCore::CleanRecycleBin(const std::string& dirName) {
+int NameSpaceToolCore::CleanRecycleBin(const std::string& dirName,
+                                       const uint64_t expireTime) {
     std::vector<FileInfo> files;
     int res = client_->ListDir(curve::mds::RECYCLEBINDIR, &files);
     if (res != 0) {
         std::cout << "List RecycleBin fail!" << std::endl;
         return -1;
     }
+
+    auto needDelete = [](const FileInfo &fileInfo,
+                         uint64_t now,
+                         uint64_t expireTime) -> bool {
+        auto filename = fileInfo.filename();
+        std::vector<std::string> items;
+        ::curve::common::SplitString(filename, "-", &items);
+
+        uint64_t dtime;
+        auto n = items.size();
+        auto id = std::to_string(fileInfo.id());
+        if (n >= 2 && items[n - 2] == id
```

Comment:
if (n >= 2 ... 
the 'n' of older file in recyclebin shuold equal 2, can delete directly. so should be n > 2 herer?

Reply:
> if (n >= 2 ...
> the 'n' of older file in recyclebin shuold equal 2, can delete directly. so should be n > 2 here?

* The `n >= 2` if only for gurantee the openration `items[n - 2]` success.
* The old format file should delete directly, because it does't have time information, we can't do any judge for it, so delete it directly.